pkg/gce-pd-csi-driver/controller.go

```diff
@@ -447,15 +448,136 @@ func (gceCS *GCEControllerServer) ControllerGetCapabilities(ctx context.Context,
 	}, nil
 }
 
+// CreatSnapshot sends a request to create snapshot to the cloud provider and then waits until the snapshot is created to return the snapshot object back.
 func (gceCS *GCEControllerServer) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	glog.Infof("CreateSnapshot called with request %v", *req)
+	name := req.GetName()
+	sourceId := req.GetSourceVolumeId()
+	volumeZone, volumeName, err := utils.SplitZoneNameId(sourceId)
+	if err != nil {
+		return nil, err
+	}
+	configuredZone := gceCS.CloudProvider.GetZone()
+	glog.Infof("volumezone %s, configuredZone %s", volumeZone, configuredZone)
+
+	if volumeZone != configuredZone {
```

Comment:
`configuredZone` in this case is just the zone in which the driver is running on. I don't think it should have any bearing on this controller call (assuming the controller service should be able to create snapshots for volumes in any zone in a multi-zonal cluster)